### ISSUE TITLE
Update to new upcoming `fax` API

### ISSUE THREAD

**Issue by gehring** [pyhf/neos#16]
We've made some pretty big changes to `fax` which will break neos. We'll be updating pypi with a new release soon but I wanted to offer to help update to the new simpler api (assuming this project is still in development). I was going to go straight for a PR but I noticed that the `neos.fit` module has some issues that I don't think I can properly fix without a bit of help.

The current implementation in `neos.fit` never allows the adam optimizer to update its state. Specifically, it always uses freshly initialized internal parameters for its scaling and momentum terms. If someone can help me better understand what `constrained_bestfit_minimized` is doing, I can submit a PR to both fix the bug and update to the new API.

**Comment by lukasheinrich** [comment]
Hi @gehring thanks so much for the offer. Yes this project is very much still active. Is there a branch on which we can preview the new API?

**Comment by gehring** [comment]
Yes, you should be able to see the new api on the master branch.

**Comment by gehring** [comment]
Oh, I should have point out that we've also updated the readme with an example which includes using a non-differentiable solver.
